### DISCUSSION TITLE
Is there a setFeatureState method?

### DISCUSSION THREAD

**Discussion by jng34** [rnmapbox/maps#3435] [Q&A]
Hi all, I was wondering how can I implement the `setFeatureState` method? 

- I created `const mapRef = useRef(null)` using the `useRef` hook and set it inside the `MapView` component, `ref={mapRef}`
- Then, I tried calling `mapRef.current.setFeatureState()` through an `onPress` callback function, but it does not work. 
- Got `TypeError: mapRef.current.setFeatureState is not a function (it is undefined), js engine: hermes` as an error. 

Any ideas or input would be greatly appreciated!

**Comment by naftalibeder** [comment]
This could be accomplished with a `FeatureCollection` stored in state, with custom properties. Something like:

```ts
const [didPress, setDidPress] = useState(false);

const featureCollection = useMemo(() => {
  return turf.featureCollection(
    features.map((f) => {
      return {
        ...f,
        properties: {
          did_press: didPress,
        }
      };
    })
  );
}, [didPress]);


return (
  <MapboxGL.ShapeSource
    shape={featureCollection}
    ...
  >
    <MapboxGL.SymbolLayer
      ...
      style={{
        iconOpacity: ['case', ['to-boolean', ['get', 'did_press']], 0.5, 1],
      }}
      aboveLayerID={mapLayerId}
    />
  </MapboxGL.ShapeSource>
);
```

Check out https://docs.mapbox.com/style-spec/reference/expressions for explanation of the style expression.
